cpp/src/binaryop/binaryop.cpp

```diff
@@ -313,8 +322,16 @@ std::unique_ptr<column> binary_operation(column_view const& lhs,
   CUDF_EXPECTS(is_fixed_width(output_type),
                "Invalid/Unsupported output datatype");
 
-  if ((lhs.type().id() == type_id::STRING) &&
-      (rhs.type().id() == type_id::STRING)) {
+  if (op == binary_operator::NULL_AWARE_EQUAL) {
+    // Check for datatype
+    CUDF_EXPECTS(lhs.type() == rhs.type(), "Both columns must be of the same type");
+    CUDF_EXPECTS(lhs.size() == rhs.size(), "Both columns must be of the same size");
```

Comment:
Could you move these restrictions to inside `null_aware_equal`.

Reply:
done.